Mathlib/GroupTheory/Subsemigroup/Centralizer.lean

```diff
@@ -121,22 +121,24 @@ theorem centralizer_subset [Mul M] (h : S ⊆ T) : centralizer T ⊆ centralizer
 
 @[to_additive addCenter_subset_addCentralizer]
 theorem center_subset_centralizer [Mul M] (S : Set M) : Set.center M ⊆ S.centralizer :=
-  fun _ hx m _ => hx m
+  fun _ hx m _ => (hx.comm m).symm
 #align set.center_subset_centralizer Set.center_subset_centralizer
 #align set.add_center_subset_add_centralizer Set.addCenter_subset_addCentralizer
 
 @[to_additive (attr := simp) addCentralizer_eq_top_iff_subset]
-theorem centralizer_eq_top_iff_subset {s : Set M} [Mul M] :
-    centralizer s = Set.univ ↔ s ⊆ center M :=
-  eq_top_iff.trans <| ⟨fun h _ hx _ => (h trivial _ hx).symm, fun h x _ _ hm => (h hm x).symm⟩
+theorem centralizer_eq_top_iff_subset {s : Set M} [Semigroup M] :
+    centralizer s = Set.univ ↔ s ⊆ center M := eq_top_iff.trans <|
+    ⟨ (fun h _ hx => Semigroup.mem_center_iff.mpr (fun _ => by rw [(h trivial _ hx)])),
+    (fun h _ _ _ hm => (h hm).comm _)⟩
```

Comment:
This indent is pretty weird, and there are way too many parens here! I think it would be better as
```suggestion
    centralizer s = Set.univ ↔ s ⊆ center M :=
  eq_top_iff.trans <| ⟨
    fun h _ hx => Semigroup.mem_center_iff.mpr fun _ => by rw [h trivial _ hx],
    fun h _ _ _ hm => (h hm).comm _⟩
```